File: backend/app/schemas/movie.py
from pydantic import BaseModel
from typing import Optional

class MovieBase(BaseModel):
    title: str
    description: str
    release_year: int
    director: str
    category: str
    image_url: Optional[str] = None

class MovieCreate(MovieBase):
    pass

class Movie(MovieBase):
    id: int
    rating: float

    class Config:
        from_attributes = True 

File: frontend/app/types/movie.ts
export interface Category {
  id: number;
  name: string;
}

export interface Movie {
  id: number;
  title: string;
  release_year: number;
  director: string;
  description: string;
  poster_url: string;
  categories: Category[];
  ratings: { rating: number }[];
}
